egs2/TEMPLATE/svs1/svs.sh

```diff
@@ -0,0 +1,1174 @@
+            done
+
+        elif [ "${feats_type}" = fbank ] || [ "${feats_type}" = stft ] ; then
+            log "Stage 2: ${feats_type} extract: data/ -> ${data_feats}/"
+
+            # Generate the fbank features; by default 80-dimensional fbanks on each frame
+            for dset in "${train_set}" "${valid_set}" ${test_sets}; do
+                if [ "${dset}" = "${train_set}" ] || [ "${dset}" = "${valid_set}" ]; then
+                    _suf="/org"
+                else
+                    _suf=""
+                fi
+                # 1. Copy datadir
+                utils/copy_data_dir.sh data/"${dset}" "${data_feats}${_suf}/${dset}"
+                # expand the utt_extra_files for multi-references
+                expand_utt_extra_files=""
+                for extra_file in ${utt_extra_files}; do
+                    for single_file in $(ls data/"${dset}"/${extra_file}); do
+                        cp ${single_file} "${data_feats}${_suf}/${dset}"
+                        expand_utt_extra_files="${expand_utt_extra_files} $(basename ${single_file})"
+                    done
+                done
+                utils/fix_data_dir.sh --utt_extra_files "${expand_utt_extra_files}" "${data_feats}${_suf}/${dset}"
+                for extra_file in ${expand_utt_extra_files}; do
+                    LC_ALL=C sort -u -k1,1 "${data_feats}${_suf}/${dset}/${extra_file}" -o "${data_feats}${_suf}/${dset}/${extra_file}"
+                done
+
+                # 2. Feature extract
+                # TODO(kamo): Wrap (nj->_nj) in make_fbank.sh
+                _nj=$(min "${nj}" "$(<${data_feats}${_suf}/${dset}/utt2spk wc -l)")
+                _opts=
+                if [ "${feats_type}" = fbank ] ; then
+                    _opts+="--fmax ${fmax} "
+                    _opts+="--fmin ${fmin} "
+                    _opts+="--n_mels ${n_mels} "
+                fi
+
+                # shellcheck disable=SC2086
+                scripts/feats/make_"${feats_type}".sh --cmd "${train_cmd}" --nj "${_nj}" \
+                    --fs "${fs}" \
+                    --n_fft "${n_fft}" \
+                    --n_shift "${n_shift}" \
+                    --win_length "${win_length}" \
+                    ${_opts} \
+                    "${data_feats}${_suf}/${dset}"
+                utils/fix_data_dir.sh "${data_feats}${_suf}/${dset}"
+
+                # 3. Derive the the frame length and feature dimension
+                scripts/feats/feat_to_shape.sh --nj "${_nj}" --cmd "${train_cmd}" \
+                    "${data_feats}${_suf}/${dset}/feats.scp" "${data_feats}${_suf}/${dset}/feats_shape"
+
+                # 4. Write feats_dim
+                head -n 1 "${data_feats}${_suf}/${dset}/feats_shape" | awk '{ print $2 }' \
+                    | cut -d, -f2 > ${data_feats}${_suf}/${dset}/feats_dim
+
+                # 5. Write feats_type
+                echo "${feats_type}" > "${data_feats}${_suf}/${dset}/feats_type"
+            done
```

Comment:
I think we can safely remove this part. There is no need for us to maintain the Kaldi frontend anymore. We can just keep using our own feature extractor `raw`.

Reply:
Removed.